core/src/net/network_thread.rs

```diff
@@ -1003,6 +1013,42 @@ mod tests {
                 .unwrap()
         );
     }
+
+    #[test]
+    fn network_thread_custom_buffer_config() -> () {
+        let addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 9788);
+        let buffer_config = BufferConfig::new(128, 13, 14, 10);
```

Comment:
This style reads very poorly, in my opinion. Having a `default` or `new` that loads the default values and then overwriting them as desired with properly named builder-style methods communicates intent a lot clearer.